### PR TITLE
Fixed #1930 Achievements activity is crashing on Android 8.1

### LINKED ISSUE #1930: Achievements activity is crashing on Android 8.1
**Summary:** 

The app crashes on clicking on the trophy icon that opens achievements.

**Steps to reproduce:** 

Click on the trophy icon. App crashes. 

Expected: Achievements activity should open.

**Add System logs:**

```
2018-10-12 12:02:28.557 9191-9191/fr.free.nrw.commons.beta E/AndroidRuntime: FATAL EXCEPTION: main
    Process: fr.free.nrw.commons.beta, PID: 9191
    java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons.beta/fr.free.nrw.commons.achievements.AchievementsActivity}: android.content.res.Resources$NotFoundException: Drawable fr.free.nrw.commons.beta:drawable/badge with resource ID #0x7f08005b
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2815)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2893)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1614)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:171)
        at android.app.ActivityThread.main(ActivityThread.java:6649)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:824)
     Caused by: android.content.res.Resources$NotFoundException: Drawable fr.free.nrw.commons.beta:drawable/badge with resource ID #0x7f08005b
     Caused by: android.content.res.Resources$NotFoundException: File res/drawable/badge.xml from drawable resource ID #0x7f08005b
        at android.content.res.ResourcesImpl.loadDrawableForCookie(ResourcesImpl.java:837)
        at android.content.res.ResourcesImpl.loadDrawable(ResourcesImpl.java:638)
        at android.content.res.MiuiResourcesImpl.loadDrawable(MiuiResourcesImpl.java:312)
        at android.content.res.Resources.getDrawableForDensity(Resources.java:891)
        at android.content.res.Resources.getDrawable(Resources.java:833)
        at android.content.Context.getDrawable(Context.java:614)
        at android.support.v4.content.ContextCompat.getDrawable(ContextCompat.java:358)
        at android.support.v7.widget.AppCompatDrawableManager.getDrawable(AppCompatDrawableManager.java:198)
        at android.support.v7.widget.AppCompatDrawableManager.getDrawable(AppCompatDrawableManager.java:186)
        at android.support.v7.content.res.AppCompatResources.getDrawable(AppCompatResources.java:100)
        at android.support.v7.widget.AppCompatImageHelper.setImageResource(AppCompatImageHelper.java:85)
        at android.support.v7.widget.AppCompatImageView.setImageResource(AppCompatImageView.java:93)
        at fr.free.nrw.commons.achievements.AchievementsActivity.onCreate(AchievementsActivity.java:123)
        at android.app.Activity.performCreate(Activity.java:7088)
        at android.app.Activity.performCreate(Activity.java:7079)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1215)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2768)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2893)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1614)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:171)
        at android.app.ActivityThread.main(ActivityThread.java:6649)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:824)
     Caused by: java.lang.IllegalArgumentException: Path string cannot be empty.
        at android.util.PathParser.nCreatePathDataFromString(Native Method)
        at android.util.PathParser.-wrap1(Unknown Source:0)
        at android.util.PathParser$PathData.<init>(PathParser.java:76)
        at android.graphics.drawable.VectorDrawable$VFullPath.updateStateFromTypedArray(VectorDrawable.java:1931)
        at android.graphics.drawable.VectorDrawable$VFullPath.inflate(VectorDrawable.java:1882)
        at android.graphics.drawable.VectorDrawable.inflateChildElements(VectorDrawable.java:741)
        at android.graphics.drawable.VectorDrawable.inflate(VectorDrawable.java:639)
        at android.graphics.drawable.DrawableInflater.inflateFromXmlForDensity(DrawableInflater.java:142)
2018-10-12 12:02:28.557 9191-9191/fr.free.nrw.commons.beta E/AndroidRuntime:     at android.graphics.drawable.Drawable.createFromXmlInnerForDensity(Drawable.java:1295)
        at android.graphics.drawable.Drawable.createFromXmlForDensity(Drawable.java:1254)
        at android.content.res.ResourcesImpl.createFromXmlForDensity(ResourcesImpl.java:1471)
        at android.content.res.ResourcesImpl.loadDrawableForCookie(ResourcesImpl.java:822)
        	... 25 more

```

**Device and Android version:** 

Redmi Note 5. Android 8.1 
 
 **Commons app version:** 

master and betaDebug/prodDebug

**Would you like to work on the issue?**

Preferably not.

### PR DESCRIPTION
**Pull request by Divyanshu2132**
Fixes #1930

Changed badge.xml file by cut shorting the pathData string as it was too log so rendering was causing the 

Tests performed

Tested on API level 27 (Asus Zenfone Max Pro M1)

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -196,5 +196,5 @@ android {
         targetCompatibility JavaVersion.VERSION_1_8
     }
 
-    buildToolsVersion buildToolsVersion
+    buildToolsVersion '28.0.3'
```
Comment (maskaravivek) [inline app/build.gradle]: Is this change related?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Sat Aug 25 23:38:03 EDT 2018
+#Mon Oct 15 20:21:03 IST 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.6-all.zip
```
Comment (maskaravivek) [inline gradle/wrapper/gradle-wrapper.properties]: Again this shouldn't be required!

File: app/build.gradle (hunk 1 above)
Comment (Divyanshu2132) [reply app/build.gradle]: No it id not related. I have just updated Build tool version and rest of dependencies to latest available. Mainly work is done in badge.xml

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (Divyanshu2132) [reply gradle/wrapper/gradle-wrapper.properties]: Sure I'll Keep that in mind

File: app/build.gradle (hunk 1 above)
Comment (maskaravivek) [reply app/build.gradle]: Yes please undo it. If you wish to update build tools version then please open a separate issue for it.

Comment (Divyanshu2132) [reply app/build.gradle]: Yes I have done the changes

**Comment by maskaravivek** [comment]
@Divyanshu2132 Looks like you have added fix for #1948 in the same PR. Please create a separate PR for it.

**Comment by maskaravivek** [comment]
Merging as the PR fixes the issue. Thanks @Divyanshu2132
